From 0555755d2de4721b00562fb821ce48811facd4bb Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Mon, 17 Feb 2025 14:59:46 +0100
Subject: [PATCH] pinctrl: stm32: Enable stm32mp2x pinctrl for aarch32

- PINCTRL_STM32MP21 is needed for stm32mp21 to boot in aarch32;
- PINCTRL_STM32MP23 is needed for stm32mp23 to boot in aarch32;
- PINCTRL_STM32MP25 is needed for stm32mp25 to boot in aarch32;

both for SoC pinctrl drivers and for hdp driver.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I4c96449f8b098e55219f5785a2fa5dd286a6e41c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/437662
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 drivers/pinctrl/stm32/Kconfig | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/pinctrl/stm32/Kconfig b/drivers/pinctrl/stm32/Kconfig
index ef6d6cfc00cd..880896e79f4e 100644
--- a/drivers/pinctrl/stm32/Kconfig
+++ b/drivers/pinctrl/stm32/Kconfig
@@ -55,13 +55,13 @@ config PINCTRL_STM32MP157
 config PINCTRL_STM32MP215
 	tristate "STMicroelectronics STM32MP215 pin control"
 	depends on OF && HAS_IOMEM
-	default MACH_STM32MP21 || ARM64
+	default MACH_STM32MP21 || ARM64 || ARM
 	select PINCTRL_STM32
 
 config PINCTRL_STM32MP235
 	tristate "STMicroelectronics STM32MP235 pin control"
 	depends on OF && HAS_IOMEM
-	default MACH_STM32MP23 || ARM64
+	default MACH_STM32MP23 || ARM64 || ARM
 	select PINCTRL_STM32
 
 config PINCTRL_STM32MP257
@@ -73,7 +73,8 @@ config PINCTRL_STM32MP257
 config PINCTRL_STM32_HDP
 	tristate "STMicroelectronics STM32 Hardware Debug Port (HDP) pin control"
 	depends on OF && HAS_IOMEM
-	default MACH_STM32MP157 || MACH_STM32MP13 || MACH_STM32MP25 || ARM64
+	default MACH_STM32MP157 || MACH_STM32MP13 || MACH_STM32MP21 || \
+		MACH_STM32MP23 || MACH_STM32MP25 || ARM64 || ARM
 	select PINMUX
 	select GENERIC_PINCONF
 	select GPIOLIB
-- 
2.50.1

